generators/chipyard/src/main/scala/harness/HarnessBinders.scala

```diff
@@ -205,35 +203,52 @@ class WithTiedOffDMI extends HarnessBinder({
   }
 })
 
-class WithSerialTLTiedOff extends HarnessBinder({
-  case (th: HasHarnessInstantiators, port: SerialTLPort) => {
-    if (DataMirror.directionOf(port.io.clock) == Direction.Input) {
-      port.io.clock := false.B.asClock
+// If tieoffs is specified, a list of serial portIds to tie off
+// If tieoffs is unspecified, ties off all serial ports
+class WithSerialTLTiedOff(tieoffs: Option[Seq[Int]] = None) extends HarnessBinder({
+  case (th: HasHarnessInstantiators, port: SerialTLPort) if (tieoffs.map(_.contains(port.portId)).getOrElse(true)) => {
+    port.io match {
+      case io: DecoupledSerialIO => io.out.ready := false.B; io.in.valid := false.B; io.in.bits := DontCare;
+      case io: SourceSyncSerialIO => {
+        io.clock_in := false.B.asClock
+        io.reset_in := false.B.asAsyncReset
+        io.in := DontCare
+        io.credit_in := DontCare
+      }
+    }
+    port.io match {
+      case io: InternalSyncSerialIO =>
+      case io: ExternalSyncSerialIO => io.clock_in := false.B.asClock
+      case _ =>
     }
-    port.io.bits.out.ready := false.B
-    port.io.bits.in.valid := false.B
-    port.io.bits.in.bits := DontCare
   }
 })
 
 class WithSimTSIOverSerialTL extends HarnessBinder({
-  case (th: HasHarnessInstantiators, port: SerialTLPort) => {
-    val bits = port.io.bits
-    if (DataMirror.directionOf(port.io.clock) == Direction.Input) {
-      port.io.clock := th.harnessBinderClock
+  case (th: HasHarnessInstantiators, port: SerialTLPort) if (port.portId == 0) => {
+    port.io match {
+      case io: InternalSyncSerialIO =>
+      case io: ExternalSyncSerialIO => io.clock_in := th.harnessBinderClock
+    }
+
+    port.io match {
+      case io: DecoupledSerialIO => {
+        // If the port is locally synchronous (provides a clock), drive everything with that clock
+        // Else, drive everything with the harnes clock
+        val clock = port.io match {
+          case io: InternalSyncSerialIO => io.clock_out
+          case io: ExternalSyncSerialIO => th.harnessBinderClock
+        }
+        withClock(clock) {
+          val ram = Module(LazyModule(new SerialRAM(port.serdesser, port.params)(port.serdesser.p)).module)
+          ram.io.ser.in <> io.out
+          io.in <> ram.io.ser.out
+
+          val success = SimTSI.connect(ram.io.tsi, clock, th.harnessBinderReset, th.p(MultiChipIdx))
```

Comment:
Should MultiChipIdx be specific to each chiptop instead of generalized to the harness?

Reply:
The code currently sets `MultiChipIdx` independently for each ChipTop (each time the HarnessBinders are applied). See lines 71-76 in HasHarnessInstantiators.scala

Reply:
I admit its a bit clunky.. I considered changing the HarnessBinder interface so they are partial functions on `th: HasHarnessInstantiators, p: Port, chipId: Int`, but that pollutes this PR, and can be done separately

Reply:
That's true, but `th.p(MultiChipIdx)` doesn't access the individual `MultiChipIdx`s for each ChipTop, it accesses `MultiChipIdx` for the testharness through th=HasHarnessInstantiators. I checked this using scala printlns and also the verilog instantiations of SimTSI both have CHIPID of 0 in the symmetric rocket config

Reply:
I also couldn't think of a nice scala solution for this. The individual chip parameters are passed in through the implicit p on ApplyHarnessBinders and it seems like it might be nontrivial to pass Parameters down through the HarnessBinder class.

Reply:
Ah, yeah, you are right. Good catch. I think the only way then is to add a new argument `chipId: Int` to the `HarnessBinderFunction` definition.